crates/cli/src/subcommands/generate/mod.rs

```diff
@@ -471,7 +472,15 @@ impl WasmCtx {
 fn format_files(generated_files: Vec<PathBuf>, lang: Language) -> anyhow::Result<()> {
     match lang {
         Language::Rust => {
-            cmd!("rustup", "component", "add", "rustfmt").run()?;
+            if !has_rust_fmt() {
+                if has_rust_up() {
+                    if let Err(err) = cmd!("rustup", "component", "add", "rustfmt").run() {
+                        return Err(anyhow::anyhow!("Failed to install rustfmt with rustup: {err}"));
```

Comment:
A better way to write this would be `.run().context("Failed to install rustfmt with Rustup")?`.